pymc/deprecator.py

```diff
@@ -0,0 +1,97 @@
+import functools
+import inspect
+import warnings
+import regex as re
+import textwrap
+
+
+def deprecator(reason=None, version=None,action='deprecate', deprecated_args=None, docs=True):
+    """
+    PyMC deprecation helper. This decorator emits deprecated warnings.
+    """    
+    if deprecated_args is not None:
+        deprecated_args = set(deprecated_args.split())
+    cause = reason
+    if cause is not None and version is not None:
+        reason = ", since version " + version + " " + "("+cause+")"
+
+    if cause is not None and version is None:
+        reason = "("+ str(cause) + ")"
+
+    if cause is None and version is not None :
+        reason = ", since version " + version
+    #this function is an edited version of the source code taken from the library Deprecated
+    #https://github.com/tantale/deprecated (MIT License)
+    def sphinxformatter(version, reason):
+        fmtsphinx = ".. deprecated:: {version}" if version else ".. deprecated::"
+        sphinxtext = [fmtsphinx.format(directive = "deprecated", version=version)]
+        width = 2 ** 16
```

Comment:
Comment magic numbers explaining why they are the way they are

Reply:
so when a para is given as reason. the existing line breaks will be used in the texts. but in case a string and not a paragraph is given then we'd need some defined line width and this renders very differently depending on the screen size because sphinx doesnt hard code the max width anymore afaik. these are the numbers the deprecated library used and its being used so i think this is safe to go?

Reply:
thanks for this comment. i had taken this from deprecated library (i saw that they had some descriptions and i figured since it was being used widely this might work) turns out. sphinx doesnt even have a fixed max width anymore. the default width for textwrap is 70. and i tested giving no width with lines and paragraphs both. works totally fine. so we dont need the unnecessary specification. im glad you asked this and i looked more into it